### ISSUE TITLE
twisted.trial.test.test_log should be ported to Python3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7812]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7812|
|Type|enhancement|
|Created|2015-03-16 06:31:40Z|
|Branch|https://github.com/twisted/twisted/tree/trialtestlog-py3-7812|

<details><summary>Searchable metadata</summary>

```
trac-id__7812 7812
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_trialtestlog_py3_7812 branches/trialtestlog-py3-7812
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1426487500107851 1426487500107851
changetime__1426490207917361 1426490207917361
version__None None
owner__hawkowl hawkowl
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [44048]) Branching to trialtestlog-py3-7812.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

Seems like it ports clean. Added to dist3, added absolute_import, spun builders.

plz reviewz

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @hawkowl|
|-|-|

LGTM; seems all the existing modules being tested are there, so it's just a matter of getting this into dist3.

Assuming the rest of the builders are happy, please land.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44050]) Merging trialtestlog-py3-7812: Port twisted.trial.test.test_log to Python 3

Author: hawkowl
Reviewer: glyph
Fixes: [#7812](https://github.com/twisted/twisted/issues/7812)
